From 137dec2957932da9cd3848ad165f3566d5879996 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Fri, 21 Dec 2018 16:50:05 -0800
Subject: [PATCH] TRITON-1060 FWAPI tests will attempt to use mockcloud CNs,
 resulting in slow and then hanging tests (avoid using mock CNs for running
 test suite)

---
 test/bin/gen-multi-test-config | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/test/bin/gen-multi-test-config b/test/bin/gen-multi-test-config
index bacf43c..064a8f9 100755
--- a/test/bin/gen-multi-test-config
+++ b/test/bin/gen-multi-test-config
@@ -49,9 +49,17 @@ image_uuid=$(curl -sS ${imgapi_url}/images'?name=sdc-smartos' | json 0.uuid)
 billing_id=$(curl -sS ${papi_url}/packages'?name=sample-128M' | json 0.uuid)
 [[ -z ${billing_id} ]] && fatal "Could not find a package named sample-128M in PAPI"
 
-
-servers=( $(curl -sS ${cnapi_url}/servers | json -a uuid setup current_platform | \
-    sort -rk3 | awk '$2 ~ /true/ { print $1 }' | head -n 2) )
+# Find two appropriate servers on which we can provision test VMs.
+# - list setup servers
+# - exclude "Virtual" servers, or Mock CNs from mockcloud because firewaller
+#   isn't (at least currently) running there
+# - only consider running servers
+# - sort to get two servers with the most recent possible platform
+servers=( $(curl -sS "${cnapi_url}/servers?extras=sysinfo&setup=true" \
+    | json -c 'this.sysinfo["System Type"] !== "Virtual"' \
+        -c 'this.status === "running"' \
+        -a uuid setup current_platform \
+    | sort -rk3 | awk '$2 ~ /true/ { print $1 }' | head -n 2) )
 
 [[ -z ${servers[0]} ]] && fatal "Could not get server1 from CNAPI"
 [[ -z ${servers[1]} ]] && fatal "Could not get server2 from CNAPI"
-- 
2.21.0

